### ISSUE TITLE
EarlGrey tests are hanging on Xcode 9 beta

### ISSUE THREAD

**Issue by liaoyuke** [google/EarlGrey#545]
By running EarlGreyFunctionalTests with EarlGrey sample app on Xcode 9 beta (both beta 1 and beta 2), tests are hanging indefinitely. The following is the log:

Showing All Messages
Test Case '-[FTRAccessibilityTest testAccessibilityElementTappedSuccessfullyWithTapAtPoint]' started.
2017-06-26 09:20:05.516896-0700 FunctionalTestRig[4965:13464724] Timer fired! Removing splash screen.
2017-06-26 09:20:05.526852-0700 FunctionalTestRig[4965:13464724] [LayoutConstraints] Sending -viewForBaselineLayout to a member of UILabel in lieu of -viewForLastBaselineLayout. Please adopt the iOS 9 API.
2017-06-26 09:20:07.977222-0700 FunctionalTestRig[4965:13464724] *** Assertion failure in -[UIViewAnimationState _transferAnimationToTrackingAnimator:], /BuildRoot/Library/Caches/com.apple.xbs/Sources/UIKit_Sim/UIKit-3683.300.2/UIView.m:828
<unknown>:0: error: -[FTRAccessibilityTest testAccessibilityElementTappedSuccessfullyWithTapAtPoint] : failed: caught "NSInternalInconsistencyException", "Attempting to transfer an animation to an animation state that is not a direct child of the animation's animation state."
(
	0   CoreFoundation                      0x0000000109ee279b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x000000010a33c121 objc_exception_throw + 48
	2   CoreFoundation                      0x0000000109ee7852 +[NSException raise:format:arguments:] + 98
	3   Foundation                          0x000000010989f612 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 193
	4   UIKit                               0x00000001071143eb -[UIViewAnimationState _transferAnimationToTrackingAnimator:] + 661
	5   UIKit                               0x0000000107d926a2 __85+[UIViewPropertyAnimator _animationBlockForTrackingAnimation:animator:trackingSetup:]_block_invoke + 103
	6   UIKit                               0x0000000107d94725 -[UIViewPropertyAnimator _runAnimations] + 222
	7   UIKit                               0x0000000107d9c802 __49-[UIViewPropertyAnimator startAnimationAsPaused:]_block_invoke_2 + 103
	8   UIKit                               0x0000000107d9c63a __49-[UIViewPropertyAnimator startAnimationAsPaused:]_block_invoke + 146
	9   UIKit                               0x0000000107d9c771 __49-[UIViewPropertyAnimator startAnimationAsPaused:]_block_invoke.1008 + 258
	10  UIKit                               0x0000000107d9bd78 -[UIViewPropertyAnimator _setupAnimationTracking:] + 128
	11  UIKit                               0x0000000107d9c305 -[UIViewPropertyAnimator startAnimationAsPaused:] + 1129
	12  UIKit                               0x0000000107d928ce +[UIViewPropertyAnimator _trackAnimationWithAnimator:forLayer:forAnimationKey:trackingSetup:] + 401
	13  UIKit                               0x0000000107d92bdb +[UIViewPropertyAnimator _trackAdditiveAnimationWithAnimator:nonAdditiveAnimation:withAnimationKey:forKeyPath:forLayer:] + 249
	14  UIKit                               0x0000000107113a5e __67-[_UIViewAdditiveAnimationAction runActionForKey:object:arguments:]_block_invoke.103 + 1255
	15  UIKit                               0x00000001071132c6 -[_UIViewAdditiveAnimationAction runActionForKey:object:arguments:] + 3186
	16  QuartzCore                          0x0000000106ed1bc3 _ZN2CA5Layer12set_positionERKNS_4Vec2IdEEb + 309
	17  QuartzCore                          0x0000000106ec27fb -[CALayer setPosition:] + 43
	18  QuartzCore                          0x0000000106ec2e90 -[CALayer setFrame:] + 549
	19  UIKit                               0x0000000107132b8d -[UIView(Geometry) setFrame:] + 368
	20  EarlGrey                            0x0000000106a31274 -[UIView(GREYAdditions) greyswizzled_setFrame:] + 308
	21  UIKit                               0x000000010706ebf6 __53-[_UINavigationParallaxTransition animateTransition:]_block_invoke.116 + 98
	22  UIKit                               0x000000010714539c +[UIView(UIViewAnimationWithBlocks) _setupAnimationWithDuration:delay:view:options:factory:animations:start:animationStateGenerator:completion:] + 560
	23  UIKit                               0x0000000107145879 +[UIView(UIViewAnimationWithBlocks) animateWithDuration:delay:options:animations:completion:] + 99
	24  EarlGrey                            0x0000000106a327de +[UIView(GREYAdditions) greyswizzled_animateWithDuration:delay:options:animations:completion:] + 158
	25  UIKit                               0x000000010706d51d -[_UINavigationParallaxTransition animateTransition:] + 2040
	26  UIKit                               0x000000010724d77a -[UINavigationController _startCustomTransition:] + 3687
	27  UIKit                               0x0000000107261751 -[UINavigationController _startDeferredTransitionIfNeeded:] + 698
	28  UIKit                               0x0000000107262ad3 -[UINavigationController __viewWillLayoutSubviews] + 140
	29  UIKit                               0x00000001074a2a98 -[UILayoutContainerView layoutSubviews] + 231
	30  UIKit                               0x000000012acc296b -[UILayoutContainerViewAccessibility layoutSubviews] + 42
	31  UIKit                               0x0000000107151e5d -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1281
	32  QuartzCore                          0x0000000106ec9026 -[CALayer layoutSublayers] + 153
	33  QuartzCore                          0x0000000106eccf51 _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 401
	34  QuartzCore                          0x0000000106ed75ee _ZN2CA5Layer28layout_and_display_if_neededEPNS_11TransactionE + 72
	35  QuartzCore                          0x0000000106e560ad _ZN2CA7Context18commit_transactionEPNS_11TransactionE + 319
	36  QuartzCore                          0x0000000106e818db _ZN2CA11Transaction6commitEv + 487
	37  QuartzCore                          0x0000000106e82498 _ZN2CA11Transaction17observer_callbackEP19__CFRunLoopObservermPv + 76
	38  CoreFoundation                      0x0000000109e85977 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
	39  CoreFoundation                      0x0000000109e858ce __CFRunLoopDoObservers + 430
	40  CoreFoundation                      0x0000000109e69c44 __CFRunLoopRun + 1332
	41  CoreFoundation                      0x0000000109e69499 CFRunLoopRunSpecific + 409
	42  EarlGrey                            0x0000000106a37016 -[GREYRunLoopSpinner grey_drainRunLoopInActiveModeForDrains:] + 326
	43  EarlGrey                            0x0000000106a36d43 -[GREYRunLoopSpinner spinWithStopConditionBlock:] + 691
	44  EarlGrey                            0x0000000106a8279e -[GREYUIThreadExecutor executeSyncWithTimeout:block:error:] + 1422
	45  EarlGrey                            0x0000000106a81e46 -[GREYUIThreadExecutor drainUntilIdleWithTimeout:] + 342
	46  EarlGrey                            0x0000000106a84443 -[GREYUIThreadExecutor grey_forcedStateTrackerCleanUp] + 51
	47  CoreFoundation                      0x0000000109e7ea3c __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
	48  CoreFoundation                      0x0000000109e7e93a _CFXRegistrationPost + 442
	49  CoreFoundation                      0x0000000109e7e682 ___CFXNotificationPost_block_invoke + 50
	50  CoreFoundation                      0x0000000109e40a02 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1826
	51  CoreFoundation                      0x0000000109e3fb7c _CFXNotificationPost + 652
	52  Foundation                          0x00000001097dd172 -[NSNotificationCenter postNotificationName:object:userInfo:] + 66
	53  EarlGrey                            0x0000000106a3e5d2 -[XCTestCase(GREYAdditions) grey_sendNotification:] + 210
	54  EarlGrey                            0x0000000106a3e4f9 -[XCTestCase(GREYAdditions) grey_tearDown] + 121
	55  XCTest                              0x0000000127711741 __51-[XCTestCase _performTearDownSequenceWithSelector:]_block_invoke + 44
	56  XCTest                              0x00000001277116fa -[XCTestCase _performTearDownSequenceWithSelector:] + 97
	57  XCTest                              0x000000012771080f __24-[XCTestCase invokeTest]_block_invoke_2 + 754
	58  XCTest                              0x0000000127757bed -[XCUITestContext performInScope:] + 187
	59  XCTest                              0x00000001277104f8 -[XCTestCase invokeTest] + 254
	60  EarlGrey                            0x0000000106a3df56 -[XCTestCase(GREYAdditions) grey_invokeTest] + 1462
	61  XCTest                              0x0000000127711424 __26-[XCTestCase performTest:]_block_invoke.362 + 42
	62  XCTest                              0x000000012775c432 +[XCTContext runInContextForTestCase:block:] + 163
	63  XCTest                              0x0000000127710dc0 -[XCTestCase performTest:] + 608
	64  XCTest                              0x000000012770d396 __27-[XCTestSuite performTest:]_block_invoke + 295
	65  XCTest                              0x000000012770cf8d -[XCTestSuite _performProtectedSectionForTest:testSection:] + 26
	66  XCTest                              0x000000012770d171 -[XCTestSuite performTest:] + 214
	67  XCTest                              0x000000012770d396 __27-[XCTestSuite performTest:]_block_invoke + 295
	68  XCTest                              0x000000012770cf8d -[XCTestSuite _performProtectedSectionForTest:testSection:] + 26
	69  XCTest                              0x000000012770d171 -[XCTestSuite performTest:] + 214
	70  XCTest                              0x000000012770d396 __27-[XCTestSuite performTest:]_block_invoke + 295
	71  XCTest                              0x000000012770cf8d -[XCTestSuite _performProtectedSectionForTest:testSection:] + 26
	72  XCTest                              0x000000012770d171 -[XCTestSuite performTest:] + 214
	73  XCTest                              0x0000000127763625 __44-[XCTTestRunSession runTestsAndReturnError:]_block_invoke + 40
	74  XCTest                              0x000000012771fb99 -[XCTestObservationCenter _observeTestExecutionForBlock:] + 480
	75  XCTest                              0x00000001277634c4 -[XCTTestRunSession runTestsAndReturnError:] + 281
	76  XCTest                              0x00000001276fcd43 -[XCTestDriver runTestsAndReturnError:] + 254
	77  XCTest                              0x000000012775b6b7 _XCTestMain + 616
	78  CoreFoundation                      0x0000000109e85dcc __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 12
	79  CoreFoundation                      0x0000000109e6a33b __CFRunLoopDoBlocks + 203
	80  CoreFoundation                      0x0000000109e69b34 __CFRunLoopRun + 1060
	81  CoreFoundation                      0x0000000109e69499 CFRunLoopRunSpecific + 409
	82  GraphicsServices                    0x000000010e2999d7 GSEventRunModal + 62
	83  UIKit                               0x00000001070860b8 UIApplicationMain + 159
	84  FunctionalTestRig                   0x00000001068ceb45 main + 165
	85  libdyld.dylib                       0x000000010c272771 start + 1
)
Test Case '-[FTRAccessibilityTest testAccessibilityElementTappedSuccessfullyWithTapAtPoint]' failed (2.754 seconds).

**Comment by khandpur** [comment]
Update: We've found fixes to both issues and testing them internally before pushing them out as PRs.

**Comment by Ruenzuo** [comment]
@khandpur thank you!, any update on the progress of this? Is there a branch in this repository we can try out before the PR is created?

**Comment by tirodkar** [comment]
@Ruenzuo I saw an issue related to Synchronization that I'm still working on providing a fix for. We'll definitely keep you updated by the end of this week with any more findings.

**Comment by steipete** [comment]
Happy to test a branch if you have an update on this.

**Comment by mettieDie** [comment]
Is there any Updates when the fixes will be pushed?

**Comment by khandpur** [comment]
closing this as it's been fixed.

**Comment by SeanChense** [comment]
```*** Assertion failure in -[UIViewAnimationState _transferAnimationToTrackingAnimator:], /BuildRoot/Library/Caches/com.apple.xbs/Sources/UIKit_Sim/UIKit-3698.33.6/UIView.m:828```
 
iOS 11.2

**Comment by khandpur** [comment]
@SeanChense which Xcode are you seeing this with?

**Comment by tirodkar** [comment]
Also, what EarlGrey version are you using?

**Comment by SeanChense** [comment]
@khandpur I am using Version 9.2 (9C40b) and EarlGrey 1.8.0

**Comment by tirodkar** [comment]
Can you please update the EarlGrey version to 1.11 or so. That is when we
added support for iOS 11.

**Comment by SeanChense** [comment]
@tirodkar Thank you.
 I thought 1.8 is the latest :). Specs in CocoaPods may be updated if convenient.

**Comment by khandpur** [comment]
On https://cocoapods.org/pods/EarlGrey it says EarlGrey 1.12.1
